File: samples/manual/node/hello_world.ts
// samples/manual/node/hello_world.ts — Minimal OCCT-Light Node.js sample.
//
// Run:
//   npx tsx samples/manual/node/hello_world.ts

import { Graph, NodeId } from "occtl";

const g = Graph.create();

const boxA: NodeId = g.makeBox({ dx: 10.0, dy: 10.5, dz: 10.0 });
console.log("Box A:", boxA);

const boxB: NodeId = g.makeBox({ dx: 10.0, dy: 10.0, dz: 10.0 });
console.log("Box B:", boxB);

const fused: NodeId = g.fuse([boxA], [boxB]);
console.log("Fused root:", fused);

console.log("Writing fused.brep...");
g.writeBrep(fused, "fused.brep");
console.log("Done.");

File: samples/manual/java/HelloWorld.java
// samples/manual/java/HelloWorld.java — Minimal OCCT-Light Java sample.
//
// Build and run (from repo root, after building the Java binding):
//   java --class-path bindings/java/build/classes/java/main:bindings/java/build/deps/*
//        samples/manual/java/HelloWorld.java

import org.occtl.*;
import java.util.List;

public class HelloWorld {
    public static void main(String[] args) {
        Graph g = Graph.create();

        NodeId boxA = g.makeBox(10.0, 10.5, 10.0);
        System.out.println("Box A: " + boxA);

        NodeId boxB = g.makeBox(10.0, 10.0, 10.0);
        System.out.println("Box B: " + boxB);

        NodeId fused = g.fuse(List.of(boxA), List.of(boxB));
        System.out.println("Fused root: " + fused);

        System.out.println("Done.");
    }
}
